src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -30,42 +30,80 @@ class XmlEncoder implements EncoderInterface, DecoderInterface, NormalizationAwa
 
     const FORMAT = 'xml';
 
+    const AS_COLLECTION = 'as_collection';
+
+    /**
+     * An array of ignored XML node types while decoding, each one of the DOM Predefined XML_* constants.
+     */
+    const DECODER_IGNORED_NODE_TYPES = 'decoder_ignored_node_types';
+
+    /**
+     * An array of ignored XML node types while encoding, each one of the DOM Predefined XML_* constants.
+     */
+    const ENCODER_IGNORED_NODE_TYPES = 'encoder_ignored_node_types';
+    const ENCODING = 'xml_encoding';
+    const FORMAT_OUTPUT = 'xml_format_output';
+
+    /**
+     * A bit field of LIBXML_* constants.
+     */
+    const LOAD_OPTIONS = 'load_options';
+    const REMOVE_EMPTY_TAGS = 'remove_empty_tags';
+    const ROOT_NODE_NAME = 'xml_root_node_name';
+    const STANDALONE = 'xml_standalone';
+    const TYPE_CASE_ATTRIBUTES = 'xml_type_cast_attributes';
+    const VERSION = 'xml_version';
+
+    private $defaultContext;
+
     /**
      * @var \DOMDocument
      */
     private $dom;
     private $format;
     private $context;
-    private $rootNodeName = 'response';
-    private $loadOptions;
-    private $decoderIgnoredNodeTypes;
-    private $encoderIgnoredNodeTypes;
 
     /**
-     * Construct new XmlEncoder and allow to change the root node element name.
-     *
-     * @param int|null $loadOptions             A bit field of LIBXML_* constants
-     * @param int[]    $decoderIgnoredNodeTypes an array of ignored XML node types while decoding, each one of the DOM Predefined XML_* Constants
-     * @param int[]    $encoderIgnoredNodeTypes an array of ignored XML node types while encoding, each one of the DOM Predefined XML_* Constants
+     * @param array $defaultContext
      */
-    public function __construct(string $rootNodeName = 'response', int $loadOptions = null, array $decoderIgnoredNodeTypes = array(XML_PI_NODE, XML_COMMENT_NODE), array $encoderIgnoredNodeTypes = array())
+    public function __construct($defaultContext = array(), int $loadOptions = null, array $decoderIgnoredNodeTypes = array(XML_PI_NODE, XML_COMMENT_NODE), array $encoderIgnoredNodeTypes = array())
     {
-        $this->rootNodeName = $rootNodeName;
-        $this->loadOptions = null !== $loadOptions ? $loadOptions : LIBXML_NONET | LIBXML_NOBLANKS;
-        $this->decoderIgnoredNodeTypes = $decoderIgnoredNodeTypes;
-        $this->encoderIgnoredNodeTypes = $encoderIgnoredNodeTypes;
+        $this->defaultContext = array(
```

Comment:
Not sure why is this whole array not defined as default directly in property definition, like rest

Reply:
Because the default values contain expressions (it's not allowed in properties definitions).

Reply:
which one? `LIBXML_NONET | LIBXML_NOBLANKS` works also as it's resolved at compile time if that's what you have in mind